pkg/config/config.go

```diff
@@ -135,3 +145,45 @@ func (rule *Rule) UnmarshalJSON(data []byte) error {
 
 	return nil
 }
+
+func (rule *Rule) MarshalYAML() (interface{}, error) {
+	result := make(map[string]any)
+	result["level"] = rule.Level
+	result["ignore"] = rule.Ignore
+
+	for key, val := range rule.Extra {
+		result[key] = val
+	}
+
+	return yaml.Marshal(&result) //nolint:wrapcheck
+}
+
+func (rule *Rule) UnmarshalYAML(value *yaml.Node) error {
+	var result map[string]any
+	if err := value.Decode(&result); err != nil {
+		return fmt.Errorf("unmarshalling rule failed %w", err)
+	}
+
+	level, ok := result["level"].(string)
+	if !ok {
+		return errLevelMustBeString
+	}
+
+	if ignore, ok := result["ignore"]; ok {
+		var dst Ignore
+
+		err := rio.YAMLRoundTrip(ignore, &dst)
+		if err != nil {
+			return fmt.Errorf("unmarshalling rule ignore failed: %w", err)
+		}
+
+		rule.Ignore = dst
+	}
+
+	delete(result, "enabled")
+
+	rule.Level = level
+	rule.Extra = result
+
+	return nil
```

Comment:
Probably not a big improvement, but could/should this part be shared with `UnmarshalJSON` to reduce code duplication?

Reply:
Fair enough! Done.